src/lib/tls/tls_extensions_cert_status_req.cpp

```diff
@@ -13,93 +13,193 @@
 #include <botan/internal/tls_reader.h>
 #include <botan/tls_exceptn.h>
 
+#include <iostream>
+
 namespace Botan::TLS
 {
 
-std::vector<uint8_t> Certificate_Status_Request::serialize(Connection_Side whoami) const
+namespace {
+class RFC6066_Empty_Certificate_Status_Request
+   {
+   public:
+      RFC6066_Empty_Certificate_Status_Request() = default;
+      RFC6066_Empty_Certificate_Status_Request(uint16_t extension_size)
+         {
+         if(extension_size != 0)
+            {
+            throw Decoding_Error("Received an unexpectedly non-empty Certificate_Status_Request");
+            }
+         }
+
+      std::vector<uint8_t> serialize() const {  return {}; }
+   };
+
+class RFC6066_Certificate_Status_Request
    {
-   std::vector<uint8_t> buf;
+   public:
+      RFC6066_Certificate_Status_Request(std::vector<uint8_t> names,
+                                         std::vector<std::vector<uint8_t>> keys)
+         : ocsp_names(std::move(names))
+         , ocsp_keys(std::move(keys)) {}
 
-   if(whoami == Connection_Side::SERVER)
-      return buf; // server reply is empty
+      RFC6066_Certificate_Status_Request(TLS_Data_Reader& reader, uint16_t extension_size)
+         {
+         if(extension_size == 0)
+            {
+            throw Decoding_Error("Received an unexpectedly empty Certificate_Status_Request");
+            }
 
-   /*
-   opaque ResponderID<1..2^16-1>;
-   opaque Extensions<0..2^16-1>;
+         const uint8_t type = reader.get_byte();
+         if(type == 1 /* ocsp */)
+            {
+            const size_t len_resp_id_list = reader.get_uint16_t();
+            ocsp_names = reader.get_fixed<uint8_t>(len_resp_id_list);
+            const size_t len_requ_ext = reader.get_uint16_t();
+            extension_bytes = reader.get_fixed<uint8_t>(len_requ_ext);
+            }
+         else
+            {
+            // RFC 6066 does not specify anything but 'ocsp' and we
+            // don't support anything else either.
+            reader.discard_next(extension_size - 1);
+            }
+         }
 
-   CertificateStatusType status_type = ocsp(1)
-   ResponderID responder_id_list<0..2^16-1>
-   Extensions  request_extensions;
-   */
+      std::vector<uint8_t> serialize() const
+         {
+         /*
+         opaque ResponderID<1..2^16-1>;
+         opaque Extensions<0..2^16-1>;
 
-   buf.push_back(1); // CertificateStatusType ocsp
+         CertificateStatusType status_type = ocsp(1)
+         ResponderID responder_id_list<0..2^16-1>
+         Extensions  request_extensions;
 
-   buf.push_back(0);
-   buf.push_back(0);
-   buf.push_back(0);
-   buf.push_back(0);
+         TODO: look into this more deeply
```

Comment:
Outstanding issue here? I don't understand what this TODO refers to

Reply:
I was wondering about the hard-coded serialization and wanted to find out what is going on here. For reference, this is now:

```C++
         // Serialization is hard-coded as we don't support advanced features
         // of this extension anyway.
         return
            {
            1,    // status_type = ocsp
            0, 0, // empty responder_id_list
            0, 0, // no extensions
            };
```